test/ash_ai/exposed_tools_test.exs

```diff
@@ -0,0 +1,180 @@
+
+  alias AshAi.Test.Music
+
+  @opts [otp_app: :ash_ai]
+
+  describe "tools filtering" do
+    test "tools: nil returns all tools (default)" do
+      tools = AshAi.exposed_tools(@opts)
+      assert length(tools) == 5
+    end
+
+    test "tools: specific list filters to only those tools" do
+      opts = Keyword.put(@opts, :tools, [:list_artists, :create_artist_after])
+
+      assert [%{name: name1}, %{name: name2}] = AshAi.exposed_tools(opts)
+      assert MapSet.new([name1, name2]) == MapSet.new([:list_artists, :create_artist_after])
+    end
+
+    test "tools: single tool in list" do
+      opts = Keyword.put(@opts, :tools, [:create_artist_manual])
+      assert [%{name: :create_artist_manual}] = AshAi.exposed_tools(opts)
+    end
+
+    test "tools: empty list returns no tools" do
+      opts = Keyword.put(@opts, :tools, [])
+      tools = AshAi.exposed_tools(opts)
+
+      assert Enum.empty?(tools)
+    end
+  end
+
+  describe "actions filtering" do
+    test "actions: specific list filters resource/action pairs" do
+      opts = Keyword.put(@opts, :actions, [{Music.ArtistAfterAction, [:read, :create]}])
+
+      # Should return only tools using read and create actions on ArtistAfterAction
+      assert [%{name: name1}, %{name: name2}] = AshAi.exposed_tools(opts)
+      assert MapSet.new([name1, name2]) == MapSet.new([:list_artists, :create_artist_after])
+    end
+
+    test "actions: wildcard returns all actions for resource" do
+      opts = Keyword.put(@opts, :actions, [{Music.ArtistAfterAction, :*}])
+
+      assert [%{name: name1}, %{name: name2}, %{name: name3}] = AshAi.exposed_tools(opts)
+
+      assert MapSet.new([name1, name2, name3]) ==
+               MapSet.new([:list_artists, :create_artist_after, :update_artist_after])
+    end
+
+    test "actions: multiple resources" do
+      opts =
+        Keyword.put(@opts, :actions, [
+          {Music.ArtistAfterAction, [:read]},
+          {Music.ArtistManual, [:create]}
+        ])
+
+      assert [%{name: name1}, %{name: name2}] = AshAi.exposed_tools(opts)
+      assert MapSet.new([name1, name2]) == MapSet.new([:list_artists, :create_artist_manual])
+    end
+
+    test "actions: raises when action not exposed as tool" do
+      assert_raise RuntimeError, "Cannot use an action that is not exposed as a tool", fn ->
+        AshAi.exposed_tools(actions: [{Music.ArtistAfterAction, [:destroy]}])
+      end
+    end
+  end
+
+  describe "exclude_actions filtering" do
+    test "exclude_actions: removes specific resource/action pairs" do
+      opts = Keyword.put(@opts, :exclude_actions, [{Music.ArtistAfterAction, :create}])
+      tools = AshAi.exposed_tools(opts)
+
+      # Should return 4 tools (all 5 except create_artist_after)
+      assert length(tools) == 4
+      tool_names = Enum.map(tools, & &1.name)
+      refute :create_artist_after in tool_names
+      assert :list_artists in tool_names
+      assert :update_artist_after in tool_names
```

Comment:
I'm personally fine with just having the refute there, but 2 asserts don't make sense in my mind, like what's with the other 2 that are in there?